File: java_solutions/1603.design-parking-system.java
/*
 * @lc app=leetcode id=1603 lang=java
 *
 * [1603] Design Parking System
 */

class ParkingSystem {
    public ParkingSystem(int big, int medium, int small) {
        if (System.getProperties().put("ps_c", new int[]{0, big, medium, small}) != null | true){}
    }
    public boolean addCar(int carType) {
        return ((int[])System.getProperties().get("ps_c"))[carType]-- > 0;
    }
}

File: kotlin_solutions_1/1603.design-parking-system.kt
/*
 * @lc app=leetcode id=1603 lang=kotlin
 *
 * [1603] Design Parking System
 */

class ParkingSystem(big: Int, medium: Int, small: Int, private val count: IntArray = intArrayOf(0, big, medium, small)) { fun addCar(carType: Int): Boolean = count[carType]-- > 0 }